search/searcher/search_geoboundingbox_test.go

```diff
@@ -198,3 +198,46 @@ func setupGeo(t *testing.T) index.Index {
 
 	return i
 }
+
+// --------------------------------------------------------------------
+
+func BenchmarkComputeGeoRangePt01(b *testing.B) {
+	onBoundary := 4
+	offBoundary := 0
+	benchmarkComputeGeoRange(b, -0.01, -0.01, 0.01, 0.01, onBoundary, offBoundary)
+}
+
+func BenchmarkComputeGeoRangePt1(b *testing.B) {
+	onBoundary := 56
+	offBoundary := 144
+	benchmarkComputeGeoRange(b, -0.1, -0.1, 0.1, 0.1, onBoundary, offBoundary)
+}
+
+func BenchmarkComputeGeoRange10(b *testing.B) {
+	onBoundary := 5464
+	offBoundary := 53704
+	benchmarkComputeGeoRange(b, -10.0, -10.0, 10.0, 10.0, onBoundary, offBoundary)
+}
+
+func BenchmarkComputeGeoRange100(b *testing.B) {
+	onBoundary := 32768
+	offBoundary := 258560
```

Comment:
One other thought is that this is a lot of on-boundary and off-boundary terms that are generated.

And, later, those terms are used in a disjunction -- so perhaps checking for a max-terms along the way would be better to short-circuit all the recursive work earlier.

Reply:
latest version has a 2nd commit which checks the max-disjunctions-clause, if any, as a part of ComputeGeoRange()